src/main/java/com/parasoft/demoapp/config/security/CustomLogoutSuccessHandler.java

```diff
@@ -1,32 +1,42 @@
 package com.parasoft.demoapp.config.security;
 
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
-import org.springframework.security.oauth2.client.oidc.web.logout.OidcClientInitiatedLogoutSuccessHandler;
-import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
 import org.springframework.security.web.authentication.logout.LogoutSuccessHandler;
 import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
 
-import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
+import java.util.Collections;
 
+@Slf4j
 @Component
 public class CustomLogoutSuccessHandler implements LogoutSuccessHandler{
 
-    private final OidcClientInitiatedLogoutSuccessHandler oidcLogoutSuccessHandler;
+    @Autowired
+    private RestTemplate restTemplate;
 
-    public CustomLogoutSuccessHandler(ClientRegistrationRepository clientRegistrationRepository) {
-        this.oidcLogoutSuccessHandler = new OidcClientInitiatedLogoutSuccessHandler(clientRegistrationRepository);
-        this.oidcLogoutSuccessHandler.setPostLogoutRedirectUri("{baseUrl}/loginPage");
-    }
+    @Value("${spring.security.oauth2.client.provider.keycloak.end-session-endpoint}")
+    private String keycloakEndSessionEndpoint;
 
     @Override
-    public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication) throws IOException, ServletException {
+    public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication) throws IOException {
         if (authentication instanceof OAuth2AuthenticationToken) {
-            oidcLogoutSuccessHandler.onLogoutSuccess(request, response, authentication);
-            return;
+            CustomOidcUser user = (CustomOidcUser) authentication.getPrincipal();
+            String idTokenValue = user.getIdToken().getTokenValue();
+            try {
+                restTemplate.getForObject(keycloakEndSessionEndpoint + "?id_token_hint={id_token_hint}",
```

Comment:
Since we can't use `spring.security.oauth2.client.provider.keycloak.issuer-uri` to get metadata for spring security, but `CustomLogoutSuccessHandler` must use issuer-uri. So we give up using `CustomLogoutSuccessHandler` instaed of using `RestTemplate` to send the logout request explicitly to Keycloak to sign out the session in Keycloak.